plugins/auth-keycloak/src/AuthPluginKeycloak.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useCallback, useMemo } from 'react';
+import Keycloak from 'keycloak-js';
+import {
+  AuthPlugin,
+  AuthPluginBase,
+  AuthPluginProps,
+} from '@deephaven/auth-plugins';
+import { useBroadcastLoginListener } from '@deephaven/jsapi-components';
+import { LoginOptions } from '@deephaven/jsapi-types';
+import { Log } from '@deephaven/log';
+
+const log = Log.module('@deephaven/js-plugin-auth-keycloak.AuthPluginKeycloak');
+
+const OIDC_AUTH_TYPE =
+  'io.deephaven.authentication.oidc.OidcAuthenticationHandler';
+
+const BASE_URL_PROPERTY = 'authentication.oidc.keycloak.url';
+const REALM_PROPERTY = 'authentication.oidc.keycloak.realm';
+const CLIENT_ID_PROPERTY = 'authentication.oidc.keycloak.clientId';
+
+/**
+ * AuthPlugin that redirects the user to the configured keycloak instance.
+ */
+function Component({
+  authConfigValues,
+  children,
+}: AuthPluginProps): JSX.Element {
+  const getConfig = useCallback(
+    (key: string) => {
+      const value = authConfigValues.get(key);
+      if (value == null) {
+        throw new Error(
+          `Keycloak config value ${key} not specified by the server`
+        );
+      }
+      return value;
+    },
+    [authConfigValues]
+  );
+
+  const keycloak = useMemo(() => {
+    const url = getConfig(BASE_URL_PROPERTY);
+    const realm = getConfig(REALM_PROPERTY);
+    const clientId = getConfig(CLIENT_ID_PROPERTY);
+    return new Keycloak({ realm, url, clientId });
+  }, [getConfig]);
+
+  const getLoginOptions = useCallback(async () => {
+    const authenticated = await keycloak.init({
+      pkceMethod: 'S256',
+      checkLoginIframe: false,
+    });
+    if (!authenticated) {
+      log.info(
+        "User isn't logged in, redirecting to IDP for authentication..."
+      );
+      keycloak.login({});
+
+      return new Promise<LoginOptions>(() => {
+        // We just want to wait for keycloak to load, so return a promise that never resolves
+      });
```

Comment:
Does `keycloak.login()` redirect? If so, add that to the comment for why this is an empty promise. But if it redirects, then do we even need to return a promise here? I guess it would prevent a potential flash or attempt to authenticate?

If not a full page redirect, we should just await it and then return, no? `keycloak.login()` is a Promise according to their types

Reply:
Yea good call - I think I just copied this from Colin's example. I'm an idiot moron who didn't proofread his changes :P